metadata-ingestion/src/datahub/ingestion/source/sql/teradata.py

```diff
@@ -678,6 +679,12 @@ def __init__(self, config: TeradataConfig, ctx: PipelineContext):
             if self.config.stateful_ingestion:
                 self.config.stateful_ingestion.remove_stale_metadata = False
 
+    def _add_default_options(self, sql_config: SQLCommonConfig) -> None:
+        """Add Teradata-specific default options"""
+        # Teradata does not support max_overflow, instead we use QueuePool when profiling
+        if sql_config.is_profiling_enabled():
+            sql_config.options.setdefault("poolclass", QueuePool)
```

Comment:
we still need to add a comment explaining what the default is, why it's insufficient, and why QueuePool is the right choice